src/ValueObject/EmailAddress.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace ThePay\ApiClient\ValueObject;
+
+class EmailAddress extends NonEmptyString
+{
+    public static function filter($value)
+    {
+        $nonEmptyString = parent::filter($value);
+        $emailAddress = filter_var($nonEmptyString, FILTER_VALIDATE_EMAIL);
```

Comment:
We use on server side email validation: `Egulias\EmailValidator\Validation\RFCValidation` from package: https://packagist.org/packages/egulias/email-validator.

We should use same validation for client SDK other wise SDK can stop request with valid email on server side or worst allow request and server will request refused with 400 error response.